_automating-workflows/api/create-workflow.md

```diff
@@ -0,0 +1,247 @@
+| :--- | :--- | :--- |
+| `workflow_id` | String | The ID of the workflow to be updated. Required for `PUT` method. |
+
+## Query parameters
+
+Workflows are normally created and provisioned in separate steps. However, once you have thoroughly tested the workflow, you can combine the create and provision steps by including the `provision` query parameter:
+
+```json
+POST /_plugins/_flow_framework/workflow?provision=true
+```
+{% include copy-curl.html %}
+
+When set to `true`, the [Provision Workflow API]({{site.url}}{{site.baseurl}}/automating-workflows/api/provision-workflow/) is executed immediately following creation. 
+
+By default, workflows are validated when they are created to ensure syntax is valid and the graph does not contain cycles. This behavior can be controlled with the `validation` query parameter. If `validation` is set to `all`, OpenSearch performs a complete template validation. Any other value of the `validation` parameter suppresses validation, allowing an incomplete/work-in-progress template to be saved. To disable template validation, set `validation` to `none`:
+
+```json
+POST /_plugins/_flow_framework/workflow?validation=none
+```
+{% include copy-curl.html %}
+
+The following table lists the available query parameters. All query parameters are optional.
+
+| Parameter | Data type | Description |
+| :--- | :--- | :--- |
+| `provision` | Boolean | Whether to provision the workflow as part of this request. Default is `false`. |
+| `validation` | String | Whether to validate the workflow. Valid values are `all` (validate the template) and `none` (do not validate the template). Default is `all`. |
+
+## Request fields
+
+The following table lists the available request fields.
+
+|Field	|Data type	|Required/Optional	|Description	|
+|:---	|:---	|:---	|:---	|
+|`name`	|String	|Required	|The name of the workflow.	|
+|`description`	|String	|Optional	|A description of the workflow.	|
+|`use_case`	|String	|Optional	|A use case, which can be used with the Search Workflow API to find related workflows. OpenSearch may provide some standard use cases in the future to ease categorization, but currently you can use this field to specify custom values.	|
+|`version`	|Object	|Optional	|A key-value map with two fields: `template` identifying the template version, and `compatibility` identifying a list of minimum required versions.	|
+|`workflows`	|Object	|Optional	|A map of workflows. Presently only the `provision` key is supported. The value for the workflow key is a key-value map which includes fields for `user_params` and lists of `nodes` and `edges`.	|
+
+#### Example Request: Register and deploy an externally hosted model (YAML)
+
+To provide a template in YAML format, specify `Content-Type: application/yaml` in the request header:
+
+```bash
+curl -XPOST "http://localhost:9200/_plugins/_flow_framework/workflow" -H 'Content-Type: application/yaml'
+```
+
+YAML templates permit comments. 
+{: .tip}
```

Comment:
For my understanding. What does `.tip` does?

Reply:
It formats the text above it with a "callout" so it looks like this in rendered form:
![Screenshot 2024-01-25 at 4 00 52 PM](https://github.com/opensearch-project/documentation-website/assets/105444904/a5ff5cbb-0d87-47db-96fc-7631bbcb5d7c)